guides/traversing-pages.rst

```diff
@@ -1,164 +1,200 @@
+on top of ``find()`` to make it easier to achieve many common use case:
 
-Named Selectors
-~~~~~~~~~~~~~~~
+``ElementInterface::has``
+    Checks whether a child element matches the given selector but without
+    returning it.
 
-Named selectors provide a way to get named XPath queries:
+``TraversableElement::findById``
+    Looks for a child element with the given id.
 
-.. code-block:: php
+``TraversableElement::findLink``
+    Looks for a link with the given text, title, id or ``alt`` attribute
+    (for images used inside links).
 
-    $selector = new \Behat\Mink\Selector\NamedSelector();
-    $handler  = new \Behat\Mink\Selector\SelectorsHandler(array(
-        'named' => $selector
-    ));
-
-    // XPath query to find the fieldset:
-    $xpath1 = $selector->translateToXPath(
-        array('fieldset', 'id|legend')
-    );
-    $xpath1 = $handler->selectorToXpath('named',
-        array('fieldset', 'id|legend')
-    );
-
-    // XPath query to find the field:
-    $xpath2 = $selector->translateToXPath(
-        array('field', 'id|name|value|label')
-    );
-    $xpath2 = $handler->selectorToXpath('named',
-        array('field', 'id|name|value|label')
-    );
-
-There's whole lot more named selectors for you to use:
-
-* ``link`` - for searching a link by its href, id, title, img alt or value
-* ``button`` - for searching a button by its name, id, value, img alt or
-  title
-* ``link_or_button`` - for searching for both, links and buttons
-* ``content`` - for searching a specific page content (text)
-* ``select`` - for searching a select field by its id, name or label
-* ``checkbox`` - for searching a checkbox by its id, name, or label
-* ``radio`` - for searching a radio button by its id, name, or label
-* ``file`` - for searching a file input by its id, name, or label
-* ``optgroup`` - for searching optgroup by its label
-* ``option`` - for searching an option by its content
-* ``table`` - for searching a table by its id or caption
-
-CSS Selectors
-~~~~~~~~~~~~~
-
-With ``Selector\CssSelector``, you can use CSS expressions to search page
-elements:
+``TraversableElement::findButton``
+    Looks for a button with the given text, title, id, ``name`` attribute
+    or ``alt`` attribute (for images used inside links).
 
-.. code-block:: php
+``TraversableElement::findField``
+    Looks for a field (``input``, ``textarea`` or ``select``) with the given
+    label, placeholder, id or ``name`` attribute.
 
-    $selector = new \Behat\Mink\Selector\CssSelector();
-    $handler  = new \Behat\Mink\Selector\SelectorsHandler(array(
-        'css' => $selector
-    ));
+.. note::
 
-    // XPath query to find the link by ID:
-    $xpath1 = $selector->translateToXPath('a#ID');
-    $xpath1 = $handler->selectorToXpath('css', 'a#ID');
+    These shortcuts are returning a single element. If you need to find all
+    matches, you will need to use ``findAll`` with the :ref:`named selector <named-selector>`.
```

Comment:
That's not entirely true. Yes, then find/findAll method needs to be used, but which selector to use is up to user. I'd prefer `,you will need to use ``findAll`` with any of [available selectors]-link to available selectors`.

Then in that section we can mentioned, that we have `named` and `css` build-in selectors with all usage examples and that a custom one can be created by implementing SelectorInterface.

Tricky part is that user have no way of registering his custom selector since Mink 2.0.